Testing-the-engine.md

```diff
@@ -0,0 +1,302 @@
+Testing the engine
+------------------
+
+Pull requests submitted to the [engine repository](https://github.com/flutter/engine)
+should be tested to prevent functional regressions. However, due to more
+diversity of environments in the engine repository, this guide is offered to
+describe how to write and run the various types of tests in the engine.
+
+## C++ - core engine
+
+If you're editing `.cc` files in https://github.com/flutter/engine/tree/master,
+you're working on the core, portable Flutter engine.
+
+### Unit tests
+
+C++ unit tests are co-located with its header and source files. For instance,
+`fml/file.h` and `fml/file.cc` have a `fml/file_unittest.cc` as sibling. When
+editing C++ files, look for its `_unittest.cc` sibling or create one if not
+present.
+
+The engine repo has a unified build system to build C, C++, Objective-C,
+Objective-C++, and Java files using [GN](https://gn.googlesource.com/gn/) and
+[Ninja](https://ninja-build.org/). Individual `_unittest.cc` files are
+referenced by the `BUILD.gn` build rule located in the folder or in an ancestor
+folder.
+
+You can run the C++ unit tests with:
+
+    `testing/run_tests.py --type=engine`
```

Comment:
Github's markdown renders this with the backquotes, since it's indented.

I'd use this instead:
```
     ```
     testing/run_tests.py --type=engine
     ```
```

Reply:
ah, thanks. Changed.